src/mlpack/tests/main_tests/linear_regression_test.cpp

```diff
@@ -18,49 +18,268 @@ static const std::string testName = "LinearRegression";
 
 using namespace mlpack;
 
-// Utility function to set a parameter and mark it as passed, using copy
-// semantics.
-template<typename T>
-void SetInputParam(const std::string& name, const T& value)
-{
-  CLI::GetParam<T>(name) = value;
-  CLI::SetPassed(name);
-}
-
-// Utility function to set a parameter and mark it as passed, using move
-// semantics.
+// Utility function to set a parameter and mark it as passed,
+// using copy semantics for lvalues and move semantics for rvalues
 template<typename T>
 void SetInputParam(const std::string& name, T&& value)
 {
-  CLI::GetParam<T>(name) = std::move(value);
+  CLI::GetParam<typename std::remove_reference<T>::type>(name)
+    = std::forward<T>(value);
```

Comment:
A minor style comment---I think that this can all fit on one line, I think it is less than 80 characters.  But maybe I am wrong (probably...); still, when you wrap, can you put the `=` on the previous line and indent the new line two tabs (four spaces) from the previous line? :)